utils/logalloc.cc

```diff
@@ -2648,6 +2652,14 @@ void region_group::update(ssize_t delta) {
     }
 }
 
+region* region_impl_to_region(region_impl* ri) {
+    return ri->_region;
+}
+
+region_impl* region_to_region_impl(region* r) {
+    return static_cast<region_impl*>(r->_impl.get());
+}
+
```

Comment:
These back-and-forth conversions are used in quite a few places in the end -- to call `occupancy()`, `evictable_occupancy()` and because `region_group` keeps -impls, not regions.

Isn't it more encapsulatish to patch `region` to provide `.occupancy()` and `.evictable_occupancy()` itself and patch `region_group` to maintain regions, not -impls?